ocf_datapipes/load/nwp/nwp.py

```diff
@@ -26,15 +26,55 @@ def __init__(
         self,
         zarr_path: Union[Path, str, list[Path], list[str]],
         provider: str = "ukv",
+        check_for_zeros: bool = False,
+        check_physical_limits: bool = False,
     ):
         """
         Opens NWP Zarr and yields it
 
         Args:
             zarr_path: Path to the Zarr file
             provider: NWP provider
+            check_for_zeros: Check for zeros in the NWP data
+            check_physical_limits: Check the physical limits of nwp data (e.g. -100<temperature<100)
         """
         self.zarr_path = zarr_path
+        self.check_for_zeros = check_for_zeros
+        self.check_physical_limits = check_physical_limits
+
+        # limits for NWP data in accordance with https://huggingface.co/openclimatefix/pvnet_uk_region/blob/main/data_config.yaml
+        self.limits = {
+            "t2m": (200, 350),  # Temperature in Kelvin (-100°C to 60°C)
+            "dswrf": (0, 1500),  # Downward short-wave radiation flux, W/m^2
+            "dlwrf": (0, 750),  # Downward long-wave radiation flux, W/m^2
+            "hcc": (0, 100),  # High cloud cover, %
+            "mcc": (0, 100),  # Medium cloud cover, %
+            "lcc": (0, 100),  # Low cloud cover, %
+            "tcc": (0, 100),  # Total cloud cover, %
+            "sde": (0, 1000),  # Snowfall depth, meters
+            "sr": (0, 10),  # Surface roughness, meters
```

Comment:
So this is getting right into OCFs data model here but @devsjc has helped me understand that we have an internal naming convention that deviates slightly from some NWP providers e.g. `sr` actually maps to [`dsrp`](https://github.com/openclimatefix/nwp-consumer/blob/5750dde11a25be6499eb97546018e5ab60ff3f67/src/nwp_consumer/internal/inputs/ecmwf/mars.py#L294) for us, not surface roughness. Obviously there would not be a way to know that as a contributor, so apologies for that. @peterdudfield FYI

Reply:
Good point! Can supply keys and then pull corresponding ranges out of consts maybe?

Reply:
you are right @Sukh-P , @peterdudfield  did help me understand the [conventions](https://huggingface.co/openclimatefix/pvnet_uk_region/blob/main/data_config.yaml#L27)